catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -524,12 +539,126 @@ public void init() {
         APPLICATION_JSON,
         (req, res) -> {
           String id = req.params(":id");
+
+          final Metacard workspace = getWorkspaceMetacard(id);
+          Attribute queries =
+              workspace.getAttribute(WorkspaceConstants.WORKSPACE_ASSOCIATED_QUERIES);
+
+          if (queries != null) {
```

Comment:
Yep so if the array of IDs is empty the delete request does fail. Added a test for this case. Note that deletion has changed a bit since your review.